src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,258 @@
+/*-------------------------------------------------------------------------
+ *
+ * role.c
+ *    Commands for ALTER ROLE statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/heapam.h"
+#include "access/htup_details.h"
+#if PG_VERSION_NUM >= 120000
+#include "access/table.h"
+#endif
+#include "catalog/catalog.h"
+#include "catalog/pg_authid.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/commands.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/deparser.h"
+#include "distributed/master_protocol.h"
+#include "distributed/worker_transaction.h"
+#include "nodes/makefuncs.h"
+#include "nodes/parsenodes.h"
+#include "utils/acl.h"
+#include "utils/builtins.h"
+#include "utils/rel.h"
+#include "utils/syscache.h"
+
+static const char * ExtractEncryptedPassword(Oid roleOid);
+static const char * CreateAlterRoleIfExistsCommand(AlterRoleStmt *stmt);
+
+/* controlled via GUC */
+bool EnableAlterRolePropagation = false;
+
+/*
+ * ProcessAlterRoleStmt actually creates the plan we need to execute for alter
+ * role statement.
+ */
+List *
+ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString)
+{
+	ListCell *optionCell = NULL;
+	List *commands = NIL;
+
+	if (!EnableAlterRolePropagation || !IsCoordinator())
+	{
+		return NIL;
+	}
+
```

Comment:
To prevent concurrent node addition, we should have something like this here:
```C
        /* don't allow concurrent node list changes that require an exclusive lock */
        LockRelationOid(DistNodeRelationId(), RowShareLock);
```

Otherwise, in one session, you can add a node, and on the other you can ALTER ROLE, and that ALTER ROLE might not get propagated to the newly added node.